pkg/api/staking.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"errors"
+	"math/big"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/staking/stakingcontract"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+)
+
+func (s *Service) stakingAccessHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !s.stakingSem.TryAcquire(1) {
+			s.logger.Debug("staking access: simultaneous on-chain operations not supported")
+			s.logger.Error(nil, "staking access: simultaneous on-chain operations not supported")
+			jsonhttp.TooManyRequests(w, "simultaneous on-chain operations not supported")
+			return
+		}
+		defer s.stakingSem.Release(1)
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+type getStakeResponse struct {
+	StakedAmount *big.Int `json:"stakedAmount"`
+}
+
+func (s *Service) stakingDepositHandler(w http.ResponseWriter, r *http.Request) {
+	overlayAddr, err := swarm.ParseHexAddress(mux.Vars(r)["address"])
+	if err != nil {
+		s.logger.Debug("get stake: decode overlay address failed", "overlay", overlayAddr, "error", err)
+		s.logger.Error(nil, "get stake: decode overlay address failed")
+		jsonhttp.BadRequest(w, "invalid address")
+		return
+	}
+
+	stakedAmount, ok := big.NewInt(0).SetString(mux.Vars(r)["amount"], 10)
+	if !ok {
+		s.logger.Error(nil, "deposit stake: insufficient amount")
+		jsonhttp.BadRequest(w, "insufficient staking amount")
+		return
```

Comment:
previous message `invalid staking amount` was correct at this point, because at this step we are converting string value to integer